.github/workflows/ci-workflow-server.yaml

```diff
@@ -0,0 +1,47 @@
+name: ci-workflow-server
+
+on:
+  push:
+    branches:
+      - main
+jobs:
+  publish:
+    permissions:
+      contents: "read"
+      id-token: "write"
+
+    if: github.event_name == 'push' && contains(github.ref, 'refs/tags/')
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18.17.0'
+
+      - name: Install dependencies
+        run: npm install
+        working-directory: ee/codegen
```

Comment:
[blocking] we have an `ee/automation` directory for this exact use case. Can we add these scripts there instead? we'll already have the octokit/rest dependency and helper for getting github token there too

Reply:
Oh didn't notice that oops.